### PR TITLE
EIP-1559 V2: Advanced gas fee modal - base fee and priority fee trends

### LINKED ISSUE #13024: EIP-1559 V2: Advanced gas fee modal - base fee and priority fee trends
Advanced gas fee modal shows upward or downwards trends for the current base fee and priority fee in the subtext below the inputs, these values are required to be made available in the redux state.

Screenshot 2021-10-25 at 3 22 21 PM

Screenshot 2021-10-25 at 4 43 29 PM

TBD: figure out the exact source of these values.

Ref: https://www.figma.com/file/DFlkrM9JRyP0hDrd4ykMZW/1559-Gas-Controls---v2.1?node-id=460%3A4673

### PR DESCRIPTION
**Pull request by NiranjanaBinoy**
Fixes: #13024 

Adding upward or downwards trends for the current base fee and priority fee in the subtext below the inputs in the advanced gas fee modal

### CODE REVIEW

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js [hunk 1]
```diff
@@ -152,13 +154,16 @@ const BaseFeeInput = () => {
 
     setBaseFeeError(error);
     setErrorValue('maxFeePerGas', error === 'editGasMaxBaseFeeImbalance');
+    setFeeTrendsValue('baseFeeTrend', baseFeeTrend);
```
[the remark this answers is not in the document]
Comment (mcmire) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js:157]: Oh, I see. Perhaps there is some discussion I am missing as well but I'm wondering if we might want to make use of "level", then? I know this is not in the designs, but what if we were to use a gray → in addition to the current ➚ and ➘?

[the remark this answers is not in the document]
Comment (NiranjanaBinoy) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js:169]: Looping in @jakehaugen here to get eyes on @mcmire 's suggestion.

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/priority-fee-input.js [hunk 2]
```diff
@@ -101,7 +107,11 @@ const PriorityFeeInput = () => {
         detailText={`≈ ${priorityFeeInFiat}`}
         numeric
       />
-      <AdvancedGasFeeInputSubtext latest="1-18 GWEI" historical="23-359 GWEI" />
+      <AdvancedGasFeeInputSubtext
+        latest="1-18 GWEI"
+        historical="23-359 GWEI"
+        feeTrend={feeTrends.priorityFeeTrend}
```
Comment (jpuri) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/priority-fee-input.js:113]: We can avoid passing `feeTrend` and rather use it from context in `AdvancedGasFeeInputSubtext`

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js:169]: Yep looks like we do not need it in state, and pass around in components. Here we are changing it on each change in input - looks weird to me.

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/priority-fee-input.js (hunk 2 above)
Comment (NiranjanaBinoy) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/priority-fee-input.js:113]: Currently, we are not passing any identifier to `AdvancedGasFeeInputSubtext` to distinguish between `baseFeeTrend` and `priorityFeeTrend`, this way we don't have to make much in `AdvancedGasFeeInputSubtext`.

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (NiranjanaBinoy) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js:169]: @jpuri and @mcmire Is there a different approach we can use here without using the state?

[the remark this answers is not in the document]
Comment (mcmire) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js]: @jakehaugen Gotcha. Thanks for the explanation!

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-input-subtext/advanced-gas-fee-input-subtext.js [hunk 3]
```diff
@@ -34,6 +34,7 @@ const AdvancedGasFeeInputSubtext = ({ latest, historical }) => {
 AdvancedGasFeeInputSubtext.propTypes = {
   latest: PropTypes.string,
   historical: PropTypes.string,
+  feeTrend: PropTypes.string,
```
Comment (mcmire) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-input-subtext/advanced-gas-fee-input-subtext.js:37]: Make sure to use `.isRequired` if this is really required 😎

**Review by mcmire — DISMISSED** [review]
One nit but looks good regardless.

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-input-subtext/advanced-gas-fee-input-subtext.js (hunk 3 above)
Comment (NiranjanaBinoy) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-input-subtext/advanced-gas-fee-input-subtext.js]: done

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/priority-fee-input.js [hunk 4]
```diff
@@ -101,7 +109,11 @@ const PriorityFeeInput = () => {
...
       />
-      <AdvancedGasFeeInputSubtext latest="1-18 GWEI" historical="23-359 GWEI" />
+      <AdvancedGasFeeInputSubtext
+        latest="1-18 GWEI"
+        historical="23-359 GWEI"
+        feeTrend={feeTrend}
+      />
     </>
   );
 };
```
Comment (jpuri) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/priority-fee-input.js:119]: Same as above.

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js [hunk 5]
```diff
@@ -200,6 +207,7 @@ const BaseFeeInput = () => {
       <AdvancedGasFeeInputSubtext
         latest={estimatedBaseFee}
         historical="23-359 GWEI"
+        feeTrend={feeTrend}
```
[the remark this answers is not in the document]
Comment (NiranjanaBinoy) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js:210]: In the subtext component, we do not have any way to identify which of the two inputs called the  `advanced-gas-fee-input-subtext.js` to decide on whether to use `baseFeeTrend` or `priorityFeeTrend`.

**Review by jpuri — DISMISSED** [review]

**Review by mcmire — DISMISSED** [review]

**Review by mcmire — APPROVED** [review]
LGTM.

**Review by jpuri — APPROVED** [review]

**Pull request merged**